### PR TITLE
Update Darwin availability annotations.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
In addition to updating the availability YAML:

* Introduces MTR_UNSTABLE_API and uses that for the metadata APIs, which are not ready to ship yet.
* Turns off MTR_PER_CONTROLLER_STORAGE_ENABLED for now.
* Updates availability on non-generated API bits.

### CODE REVIEW

File: src/darwin/Framework/CHIP/templates/availability.yaml [hunk 1]
```diff
@@ -7803,8 +7803,12 @@
               Feature:
                   DeadFrontBehavior: DeadFront
 
-- release: "Future"
-  versions: "future"
+- release: "Winter 2024"
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/templates/availability.yaml:7806]:
```suggestion
- release: "Spring 2024"
```
Should this be Spring?

File: src/darwin/Framework/CHIP/templates/MTRClusters.zapt [hunk 2]
```diff
@@ -35,7 +35,7 @@ NS_ASSUME_NONNULL_BEGIN
 {{#if (isInConfigList
        (concat (asUpperCamelCase cluster preserveAcronyms=true) "::" (asUpperCamelCase command preserveAcronyms=true))
        "LegacyCommandsWithOnlyOptionalArguments")}}
-{{availability cluster command=command minimalRelease="Future"}}; {{! TODO: Use the right thing here when we know what it's called }}
+{{availability cluster command=command minimalRelease="Winter 2024"}};
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/templates/MTRClusters.zapt:38]:
```suggestion
{{availability cluster command=command minimalRelease="Spring 2024"}};
```
Should this be Spring?

File: src/darwin/Framework/CHIP/templates/MTRBaseClusters.zapt [hunk 3]
```diff
@@ -40,7 +40,7 @@ NS_ASSUME_NONNULL_BEGIN
 {{#if (isInConfigList
        (concat (asUpperCamelCase cluster preserveAcronyms=true) "::" (asUpperCamelCase command preserveAcronyms=true))
        "LegacyCommandsWithOnlyOptionalArguments")}}
-{{availability cluster command=command minimalRelease="Future"}}; {{! TODO: Use the right thing here when we know what it's called }}
+{{availability cluster command=command minimalRelease="Winter 2024"}};
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/templates/MTRBaseClusters.zapt:43]:
```suggestion
{{availability cluster command=command minimalRelease="Spring 2024"}};
```
Should this be Spring?

File: src/darwin/Framework/CHIP/templates/availability.yaml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/templates/availability.yaml]: Per Slack discussion: "Early 2024"

File: src/darwin/Framework/CHIP/templates/MTRClusters.zapt (hunk 2 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/templates/MTRClusters.zapt]: "Early 2024"

File: src/darwin/Framework/CHIP/templates/MTRBaseClusters.zapt (hunk 3 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/templates/MTRBaseClusters.zapt]: "Early 2024"

**Review by andy31415 — APPROVED** [review]

**Review by jtung-apple — APPROVED** [review]

**Pull request merged**